src/datasource.js

```diff
@@ -246,6 +277,40 @@ export class GenericDatasource {
     return graphs;
   }
 
+  /**
+   * Transform data returned by time series query into Grafana annotation format.
+   * @param results
+   * @param options
+   * @returns Array
+   */
+  static transformAnnotationResults(results, timeField, timeEndField, textField) {
+    const annotations = [];
+    results.forEach((result) => {
+      const responseObject = {};
+      responseObject.time = Number(result[timeField]) / 1000000;
+      if (!responseObject.time && result.attributes) {
```

Comment:
It seems like there may be a logic error:
```
responseObject.time = Number(result[timeField]) / 1000000;
 if (!responseObject.time && result.attributes) {
```
If `timeField` is not in `result`, then `responseObject.time = NaN`.  Then `!responseObject.time` will be `false` b/c `responseObject.time` is not `undefined` but rather `NaN`.

It may make the logic cleaner to do safety checks for `if (result[timeField]) ... else if (result.attributes && result.attributes[timeField]`

Reply:
`NaN` is also falsey so I think this still works as is, unless I misunderstood what you mean.

Reply:
Actually so is `0` so the `time === 0` case is covered here as well.